### ISSUE TITLE
Error in 【MSBuildWorkspace.Create() .OpenSolutionAsync()】：System.Exception (cannotfound BuildHost)

### ISSUE THREAD

**Issue by Miltomo** [dotnet/roslyn#74070]
**Version Used**: 17.11.0 Preview 2.1

**Error Message**:
Unable to find build host in “c:\program files\microsoft visual studio\2022\preview\common7\ide\commonextensions\microsoft\vbcsharp\languageservices\BuildHost-netcore\Microsoft.CodeAnalysis.Workspaces.MSBuild.BuildHost.dll”

**Steps to Reproduce**:
1. I am developing VS extensions 【VSIX Project】
2. I'm just using Microsoft.CodeAnalysis.MSBuild to get a Solution var
3. I have tried many methods (such as using **Locator** to register, but in reality, Locator will report an error saying that an instance already exists), but this error still occurs
4. In fact, I checked the folder and found that the **"BuildHost-netcore"** folder does not exist under **_..\Commonextensions\Microsoft\VBCSharp\LanguageServices_**
5. **Annoying BuildHost, annoying bad code**,  I think I have known why no one want to develop VS extensions

**StackTrace**:
在 Microsoft.CodeAnalysis.MSBuild.BuildHostProcessManager.AssertBuildHostExists(String buildHostPath)
   在 Microsoft.CodeAnalysis.MSBuild.BuildHostProcessManager.GetNetCoreBuildHostPath()
   在 Microsoft.CodeAnalysis.MSBuild.BuildHostProcessManager.CreateDotNetCoreBuildHostStartInfo()
   在 Microsoft.CodeAnalysis.MSBuild.BuildHostProcessManager.<GetBuildHostAsync>d__9.MoveNext()
   在 System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   在 System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   在 Microsoft.CodeAnalysis.MSBuild.BuildHostProcessManager.<GetBuildHostWithFallbackAsync>d__8.MoveNext()
   在 System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   在 System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   在 Microsoft.CodeAnalysis.MSBuild.MSBuildProjectLoader.Worker.<LoadProjectFileInfosAsync>d__20.MoveNext()
   在 System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   在 System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   在 Microsoft.CodeAnalysis.MSBuild.MSBuildProjectLoader.Worker.<LoadProjectInfosFromPathAsync>d__21.MoveNext()
   在 System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   在 Microsoft.CodeAnalysis.MSBuild.MSBuildProjectLoader.Worker.<LoadAsync>d__19.MoveNext()
   在 System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   在 System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   在 Microsoft.CodeAnalysis.MSBuild.MSBuildProjectLoader.<LoadSolutionInfoAsync>d__23.MoveNext()
   在 System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   在 Microsoft.CodeAnalysis.MSBuild.MSBuildProjectLoader.<LoadSolutionInfoAsync>d__23.MoveNext()
   在 System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   在 System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   在 Microsoft.CodeAnalysis.MSBuild.MSBuildWorkspace.<OpenSolutionAsync>d__25.MoveNext()
   在 System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   在 System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   在 System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()

**Version Info**:
C:\Users\Administrator>dotnet --info
.NET SDK:
 Version:           8.0.302
 Commit:            ef14e02af8
 Workload version:  8.0.300-manifests.c3d67e2f
 MSBuild version:   17.10.4+10fbfbf2e

**Comment by kamronbatman** [comment]
Yeah I am getting the same error. This seems to be a combination of the bump to 4.9.0 or 4.10.0, and 8.0.302.
On my CI/CD (Azure Pipelines, newest windows and .NET 8.0.302), it never finishes finding the solution and the program hangs indefinitely.

**Comment by tobyash86** [comment]
If you are developing VS extension consider using VisualStudioWorkspace instead as mentioned in #73854

**Comment by kamronbatman** [comment]
> If you are developing VS extension consider using VisualStudioWorkspace instead as mentioned in [#73854](https://github.com/dotnet/roslyn/issues/73854)

In my particular case I am developing a dotnet tool. I am still encountering this issue.

**Comment by prochan2** [comment]
Same as @kamronbatman here. See https://github.com/postsharp/PostSharp.Engineering/blob/63aae72db118c43a2d7bdd56f0f093f2ce3b8bad/src/PostSharp.Engineering.BuildTools/XmlDoc/RemoveInternalsCommand.cs#L26

**Comment by cirrusqm** [comment]
I also have the same issue, and VisualStudioWorkspace won't work for me.

**Comment by kamronbatman** [comment]
My particular issue has been fixed here: https://github.com/dotnet/roslyn/issues/75967

**Comment by JoeRobich** [comment]
This issue is resolved by https://github.com/dotnet/roslyn/pull/76747
